train_and_eval/train/train_imagenet.py

```diff
@@ -0,0 +1,607 @@
+import torch as ch
+from torch.cuda.amp import GradScaler
+from torch.cuda.amp import autocast
+import torch.nn.functional as F
+import torch.distributed as dist
+ch.backends.cudnn.benchmark = True
+ch.autograd.profiler.emit_nvtx(False)
+ch.autograd.profiler.profile(False)
+
+from torchvision import models
+import torchmetrics
+import numpy as np
+from tqdm import tqdm
+
+import os
+import time
+import json
+from uuid import uuid4
+from typing import List
+from pathlib import Path
+from argparse import ArgumentParser
+
+from fastargs import get_current_config
+from fastargs.decorators import param
+from fastargs import Param, Section
+from fastargs.validation import And, OneOf
+
+from ffcv.pipeline.operation import Operation
+from ffcv.loader import Loader, OrderOption
+from ffcv.transforms import ToTensor, ToDevice, Squeeze, NormalizeImage, \
+    RandomHorizontalFlip, ToTorchImage
+from ffcv.fields.rgb_image import CenterCropRGBImageDecoder, \
+    RandomResizedCropRGBImageDecoder
+from ffcv.fields.basics import IntDecoder
+from MRL import *
+
+Section('model', 'model details').params(
+    arch=Param(And(str, OneOf(models.__dir__())), default='resnet18'),
+    pretrained=Param(int, 'is pretrained? (1/0)', default=0),
+    efficient=Param(int, "MRL-E?", default=0),
+    mrl=Param(int, "MRL?", default=0),
+    nesting_start=Param(int, '2**i will be starting dimension for nesting', default=3),
+    fixed_feature=Param(int, 'In case we want to do the fixed feature training, by default it is 2048', default=2048)
+)
+
+Section('resolution', 'resolution scheduling').params(
+    min_res=Param(int, 'the minimum (starting) resolution', default=160),
+    max_res=Param(int, 'the maximum (starting) resolution', default=160),
+    end_ramp=Param(int, 'when to stop interpolating resolution', default=0),
+    start_ramp=Param(int, 'when to start interpolating resolution', default=0)
+)
+
+Section('data', 'data related stuff').params(
```

Comment:
remove all "stuff" words in Section descriptions and just write "data parameters"

Reply:
That is directly taken from FFCV.